main.go

```diff
@@ -22,51 +22,74 @@ func main() {
 
 	// Iterate over each pull request
 	for _, pr := range prs {
-		fmt.Printf("PR #%d: %s\n", pr.GetNumber(), pr.GetTitle())
-
-		fmt.Printf("Getting diff for PR #%d\n", pr.GetNumber())
-		diff, err := client.GetPrDiff(pr)
-		if err != nil {
-			log.Fatalf("Failed to get raw diff: %v", err)
+		comments := reviewPR(pr, client)
+		commitId := pr.GetHead().GetSHA()
+		for _, comment := range comments {
+			// post to github
+			fmt.Printf("Posting comment to PR #%d\n", pr.GetNumber())
+			ghComment := client.ParsedCommentToGithubComment(comment, commitId)
+			err := client.PostComment(pr, ghComment)
+			if err != nil {
+				fmt.Printf("Failed to post comment: %v\n", err)
+				fmt.Printf("Comment: %v\n", ghComment)
+			}
 		}
+	}
+}
+
+func reviewPR(pr *github.PullRequest, client github_client.GithubRepoClient) []*comments.Comment {
+	fmt.Printf("PR #%d: %s\n", pr.GetNumber(), pr.GetTitle())
+
+	fmt.Printf("Getting diff for PR #%d\n", pr.GetNumber())
+	diff, err := client.GetPrDiff(pr)
+	if err != nil {
```

Comment:
Suggestion: consider handling error gracefully instead of calling `log.Fatal`
DEBUG INFO
[File] main.go

[Side] RIGHT

[Start Line] 45

[End Line] 0

[Raw Comment] [+ Line 45] Suggestion: consider handling error gracefully instead of calling `log.Fatal` 
[
[Original Code]  a/main.go b/main.go
index 27a516a..5dbc015 100644
 package main
 
 import (
4 -	"encoding/json"
 	"fmt"
6 -	"io/ioutil"
 	"log"
 
9 -	"github.com/birdsean/review-droid/github"
7 +	"github.com/birdsean/review-droid/comments"
8 +	"github.com/birdsean/review-droid/github_client"
 	"github.com/birdsean/review-droid/openai"
 	"github.com/birdsean/review-droid/transformer"
11 +	"github.com/google/go-github/v53/github"
 )
 
 func main() {
15 -	client := github.GithubRepoClient{}
15 +	client := github_client.GithubRepoClient{}
 	client.Init()
 
 	prs, err := client.GetPrs()
 
 	// Iterate over each pull request
 	for _, pr := range prs {
25 -		fmt.Printf("PR #%d: %s\n", pr.GetNumber(), pr.GetTitle())
26 -
27 -		fmt.Printf("Getting diff for PR #%d\n", pr.GetNumber())
28 -		diff, err := client.GetPrDiff(pr)
29 -		if err != nil {
30 -			log.Fatalf("Failed to get raw diff: %v", err)
25 +		comments := reviewPR(pr, client)
26 +		commitId := pr.GetHead().GetSHA()
27 +		for _, comment := range comments {
28 +			// post to github
29 +			fmt.Printf("Posting comment to PR #%d\n", pr.GetNumber())
30 +			ghComment := client.ParsedCommentToGithubComment(comment, commitId)
31 +			err := client.PostComment(pr, ghComment)
32 +			if err != nil {
33 +				fmt.Printf("Failed to post comment: %v\n", err)
34 +				fmt.Printf("Comment: %v\n", ghComment)
35 +			}
 		}
37 +	}
38 +}
39 +
40 +func reviewPR(pr *github.PullRequest, client github_client.GithubRepoClient) []*comments.Comment {
41 +	fmt.Printf("PR #%d: %s\n", pr.GetNumber(), pr.GetTitle())
42 +
43 +	fmt.Printf("Getting diff for PR #%d\n", pr.GetNumber())
44 +	diff, err := client.GetPrDiff(pr)
45 +	if err != nil {
46 +		log.Fatalf("Failed to get raw diff: %v", err)
47 +	}
48 +
49 +	diffTransformer := transformer.DiffTransformer{}
50 +	diffTransformer.Transform(diff)
 
33 -		diffTransformer := transformer.DiffTransformer{}
34 -		diffTransformer.Transform(diff)
52 +	fileSegments := diffTransformer.GetFileSegments()
53 +	allComments := []*comments.Comment{}
54 +	// failedComments := []string{}
 
36 -		segments := diffTransformer.GetSegments()
37 -		allComments := []string{}
38 -		fmt.Printf("Getting commments for %d segments\n", len(segments))
56 +	fmt.Printf("Getting comments for %d segments\n", len(fileSegments))
57 +	for filename, segments := range fileSegments {
 		for _, segment := range segments {
 			comment := retrieveComments(segment)
41 -			allComments = append(allComments, comment)
60 +			if comment == nil {
61 +				// failedComments = append(failedComments, segment)
62 +				continue
63 +			}
64 +			zippedComments, err := comments.ZipComment(segment, *comment, filename)
65 +			if err != nil {
66 +				log.Fatalf("Failed to zip comment: %v", err)
67 +			}
68 +			allComments = append(allComments, zippedComments...)
69 +			if len(allComments) >= 1 {
70 +				break
71 +			}
72 +		}
73 +		if len(allComments) >= 1 {
74 +			break
 		}
43 -
44 -		writeResults(allComments)
 	}
77 +
78 +	return allComments
 }
 
48 -func retrieveComments(segment string) string {
81 +func retrieveComments(segment string) *string {
 	openAiClient := openai.OpenAiClient{}
 	openAiClient.Init()
 	completion, err := openAiClient.GetCompletion(segment)
 		fmt.Printf("Failed to get completion: %v\n", err)
 	}
 
56 -	fmt.Println("********************")
57 -	fmt.Println(*completion)
58 -	fmt.Println("********************")
59 -	return *completion
60 -}
61 -
62 -func writeResults(comments []string) {
63 -	fileContents, err := json.Marshal(comments)
64 -	if err != nil {
65 -		log.Fatalf("Failed to marshal comments: %v", err)
66 -	}
67 -	// save fileContents to results.json
68 -	err = ioutil.WriteFile("results.json", fileContents, 0644)
69 -	if err != nil {
70 -		log.Fatalf("Failed to write comments to file: %v", err)
89 +	if completion != nil {
90 +		fmt.Println("********************")
91 +		fmt.Println(*completion)
92 +		fmt.Println("********************")
 	}
94 +	return completion
 }